projects/npm-tools/packages/npm-scripts/src/utils/getExternalExportsWebpackConfigs.js

```diff
@@ -0,0 +1,246 @@
+/**
+ * SPDX-FileCopyrightText: © 2019 Liferay, Inc. <https://liferay.com>
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+/* eslint-disable @liferay/no-dynamic-require */
+
+const MiniCssExtractPlugin = require('mini-css-extract-plugin');
+const path = require('path');
+const resolve = require('resolve');
+const TerserPlugin = require('terser-webpack-plugin');
+
+const convertImportsToExternals = require('./convertImportsToExternals');
+const createTempFile = require('./createTempFile');
+
+module.exports = function getExternalExportsWebpackConfigs(
+	projectDir,
+	buildConfig
+) {
+	const {exports, imports} = buildConfig;
+
+	const allExternals = convertImportsToExternals(imports, 3);
+
+	return exports
+		.filter((exportsItem) => !exportsItem.path.startsWith('.'))
+		.reduce((webpackConfigs, exportsItem) => {
+			const importPath = getEntryImportPath(exportsItem);
+
+			const externals = {
+				...allExternals,
+			};
+
+			delete externals[exportsItem.path];
+
+			const webpackConfig = {
+				entry: {
+					[exportsItem.name]: {
+						import: importPath,
+					},
+				},
+				experiments: {
+					outputModule: true,
+				},
+				externals,
+				externalsType: 'module',
+				module: {
+					rules: [
+						{
+							exclude: /node_modules/,
+							test: /\.js$/,
```

Comment:
Will we need to account for .ts(x) files or .jsx files? I see you are doing that in the other two webpack config files